clients/admin-ui/src/features/system/form.ts

```diff
@@ -91,4 +96,12 @@ export const transformFormValuesToSystem = (formValues: FormValues): System => {
         ? undefined
         : formValues.administrating_department,
   };
+  if (features.plus) {
+    if (features.dictionaryService && formValues?.meta?.vendor?.id) {
+      payload.meta = {};
+      payload.meta.vendor = {};
+      payload.meta.vendor.id = formValues.meta.vendor.id;
```

Comment:
would this work?
```suggestion
      payload.meta = { vendor: { id: formValues.meta.vendor.id } };
```

Reply:
I think this needs to be handled a little differently. both of our impls will overwrite everything that is within meta. So something like `meta.connector` would be deleted. I'll update it to handle that scenario.